### PR TITLE
Sofies Happy Thought project

### PR DESCRIPTION
**Pull request by SofieFerrari**
## Netlify link
https://bucolic-beignet-3163b4.netlify.app

### CODE REVIEW

File: src/components/LikeThought.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import './Thoughts.css'
+import { GetThoughts } from './Thoughts'
+
+export const LikeThoughts = ({ index }) => {
+	const [likes, setLikes] = useState(index.hearts)
```
Comment (maikanetaka) [inline src/components/LikeThought.jsx:7]: I think your tab indentation setting in VS code is now four-space. Think better to change two-space instead to align the standard to avoid unnecessary changes when you work with someone :)

File: src/components/Thoughts.jsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	}, [])
+	//fixa error här
+	const handleFormSubmit = (event) => {
+		event.preventDefault()
+	}
+
+	//filtrera ut det som ska visas
+	return (
+		<>
+			{thoughts.map((index) => (
```
Comment (maikanetaka) [inline src/components/Thoughts.jsx:31]: Nice map function!

File: src/components/LikeThought.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import './Thoughts.css'
+import { GetThoughts } from './Thoughts'
+
+export const LikeThoughts = ({ index }) => {
```
Comment (maikanetaka) [inline src/components/LikeThought.jsx:6]: Maybe this prop passed to LikeThoughts is named index is a bit misleading. Consider others, like "thought" might be better I think.

File: src/components/LikeThought.jsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	const [likes, setLikes] = useState(index.hearts)
+
+const handleLike = (index) => {
```
Comment (maikanetaka) [inline src/components/LikeThought.jsx:9]: Similarly, the parameter for handleLike is named index, which could be renamed to reflect that it's a thought object :)

File: src/components/LikeThought.jsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react'
```
Comment (maikanetaka) [inline src/components/LikeThought.jsx:1]: Imported useEffect but seems like it's not used in the component. so better to remove to keep the code clean :)

File: src/components/Thoughts.jsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+				})
+			//fixa error här
+		}
+	}, [newThought, inputReady])
+
+	const handleSubmit = (event) => {
+		event.preventDefault()
+		if (newThought.length < 5 || newThought.length > 140) {
+			setError(
+				'Please typ something with at least 5 letters and with the most 140 letters'
```
Comment (maikanetaka) [inline src/components/Thoughts.jsx:75]: Friendly error message showing! :)

File: src/components/Thoughts.jsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+		} else if (newThought !== '') {
+			setInputReady(true)
+		}
+	}
+
+	return (
+		<div className="input-wrapper">
+			<label>What is making you happy right now?</label>
+			<form onSubmit={handleSubmit}>
+				<input
```
Comment (maikanetaka) [inline src/components/Thoughts.jsx:86]: great job of implementing a controlled input like this!

**Review by maikanetaka — COMMENTED** [review]
Hej Sofie! Your code demonstrates a good grasp of React fundamentals, including component creation, state management, effect hooks, and interacting with an API!
Great job!! 🌟

**Review by dzc1 — APPROVED** [review]
Hey Sofie, 

Although the project meets all the requirements and indeed everything looks great, I just have question to ask you, both `GetThoughts` and `PostThoughts` are being fetched from 1 file where as you developed it works perfectly but my question is simple, are you doing this for organisation ? .

Nevertheless, the application is perfect! 

Keep up the great work!

**Review by AntonellaMorittu — APPROVED** [review]
